api/buckets/service.go

```diff
@@ -2003,6 +2025,47 @@ func (s *Service) PullPathAccessRoles(ctx context.Context, req *pb.PullPathAcces
 	}, nil
 }
 
+func (s *Service) DefaultArchiveConfig(ctx context.Context, req *pb.DefaultArchiveConfigRequest) (*pb.DefaultArchiveConfigResponse, error) {
+	log.Debug("received set default archive config request")
+
+	if !s.Buckets.IsArchivingEnabled() {
+		return nil, ErrArchivingFeatureDisabled
+	}
+
+	ba, err := s.Collections.BucketArchives.GetOrCreate(ctx, req.Key)
+	if err != nil {
+		return nil, fmt.Errorf("getting bucket archive data: %s", err)
+	}
+	archiveConfig := ba.DefaultArchiveConfig
+	if archiveConfig == nil {
+		// The queried BucketArchive was created before we added .DefaultArchiveConfig, so just return the default
+		archiveConfig = &defaultDefaultArchiveConfig
+	}
+	return &pb.DefaultArchiveConfigResponse{
+		ArchiveConfig: toPbArchiveConfig(archiveConfig),
+	}, nil
+}
+
+func (s *Service) SetDefaultArchiveConfig(ctx context.Context, req *pb.SetDefaultArchiveConfigRequest) (*pb.SetDefaultArchiveConfigResponse, error) {
+	log.Debug("received set default archive config request")
+
+	if !s.Buckets.IsArchivingEnabled() {
+		return nil, ErrArchivingFeatureDisabled
+	}
+
+	ba, err := s.Collections.BucketArchives.GetOrCreate(ctx, req.Key)
+	if err != nil {
+		return nil, fmt.Errorf("getting bucket archive data: %s", err)
+	}
+
+	ba.DefaultArchiveConfig = fromPbArchiveConfig(req.ArchiveConfig)
+	err = s.Collections.BucketArchives.Replace(ctx, ba)
```

Comment:
If the user calls `SetDefaultArchiveConfig` explicitly, we do persist the `ArchiveConfig` they provide and it will be used as the default in `DefaultArchiveConfig` and `Archive` from now on.